tests/integration/actions/ee_details/base.py

```diff
@@ -0,0 +1,129 @@
+from ....defaults import FIXTURES_DIR
+
+
+TEST_FIXTURE_DIR = os.path.join(FIXTURES_DIR, "integration/actions/ee_details")
+TEST_CONFIG_FILE = os.path.join(TEST_FIXTURE_DIR, "ansible-navigator.yml")
+
+
+class Command(NamedTuple):
+    """command details"""
+
+    execution_environment: bool
+    container_engine: str = container_runtime_or_fail()
+    command: str = "ansible-navigator"
+    mode: str = "interactive"
+    subcommand: Union[None, str] = None
+
+    def join(self):
+        """create cli command"""
+        args = []
+        args.append(self.command)
+        if self.subcommand is not None:
+            args.append(self.subcommand)
+        args.extend(["--ee", self.execution_environment])
+        args.extend(["--ce", self.container_engine])
+        args.extend(["--mode", self.mode])
+        return " ".join(shlex.quote(str(arg)) for arg in args)
+
+
+class Step(NamedTuple):
+    # pylint: disable=too-few-public-methods
+    """test data object"""
+    user_input: str
+    comment: str
+
+    look_fors: List[str] = []
+    playbook_status: Union[None, str] = None
+    step_index: int = 0
+    wait_on_cli_prompt: bool = False
+
+
+def add_indicies(steps):
+    """update the index of each"""
+    return (step._replace(step_index=idx) for idx, step in enumerate(steps))
+
+
+def step_id(value):
+    """return the test id from the test step object"""
+    return f"{value.step_index}-{value.user_input}-{value.comment}"
+
+
+base_steps = (
+    Step(user_input=":0", comment="Check gather"),
+    Step(user_input=":back", comment="Return to play list"),
+    Step(user_input=":1", comment="Check report"),
+    Step(user_input=":back", comment="Return to playlist"),
+    Step(user_input=":2", comment="Report tasks"),
+    Step(user_input=":0", comment="View report", look_fors=["ansible_version", "base_os"]),
+)
+
+
+class BaseClass:
+    """base class for interactive stdout tests"""
+
+    UPDATE_FIXTURES = False
+
+    @staticmethod
+    @pytest.fixture(scope="module", name="tmux_session")
+    def fixture_tmux_session(request):
+        """tmux fixture for this module"""
+        params = {
+            "window_name": request.node.name,
+            "setup_commands": [
+                "export ANSIBLE_DEVEL_WARNING=False",
+                "export ANSIBLE_DEPRECATION_WARNINGS=False",
+            ],
+            "config_path": TEST_CONFIG_FILE,
+            "pane_height": "1000",
+        }
+        with TmuxSession(**params) as tmux_session:
+            yield tmux_session
+
+    def test(self, request, tmux_session, step):
+        # pylint:disable=unused-argument
+        # pylint: disable=too-few-public-methods
+        # pylint: disable=too-many-arguments
+
+        """test"""
+        assert os.path.exists(TEST_CONFIG_FILE)
+
+        received_output = tmux_session.interaction(
+            value=step.user_input,
+            wait_on_cli_prompt=step.wait_on_cli_prompt,
+            wait_on_playbook_status=step.playbook_status,
```

Comment:
Have updated the method in PR https://github.com/ansible/ansible-navigator/pull/254 to make it bit more cleaner and added a timeout logic as well.